### ISSUE TITLE
ResNet50 to one single Colaboratory file. Train the model and determine the type 1 and type 5 errors for testing.

### ISSUE THREAD

**Issue by EKD01** [weiaicunzai/pytorch-cifar100#91]
# Import necessary libraries
import tensorflow as tf
from tensorflow.keras.applications import ResNet50
from tensorflow.keras.preprocessing.image import ImageDataGenerator
from tensorflow.keras.optimizers import Adam
from tensorflow.keras.losses import SparseCategoricalCrossentropy
from tensorflow.keras.metrics import SparseTopKCategoricalAccuracy
import numpy as np

# Load the CIFAR-10 dataset
(x_train, y_train), (x_test, y_test) = tf.keras.datasets.cifar10.load_data()

# Normalize the data
x_train = x_train.astype('float32') / 255.0
x_test = x_test.astype('float32') / 255.0

# Define the model
model = ResNet50(weights=None, input_shape=(32, 32, 3), classes=10)

# Compile the model
model.compile(
    optimizer=Adam(),
    loss=SparseCategoricalCrossentropy(),
    metrics=[SparseTopKCategoricalAccuracy(k=1, name='top_1_accuracy'),
             SparseTopKCategoricalAccuracy(k=5, name='top_5_accuracy')]
)

# Data augmentation
datagen = ImageDataGenerator(
    width_shift_range=0.1,
    height_shift_range=0.1,
    horizontal_flip=True
)
datagen.fit(x_train)

# Train the model
batch_size = 64
epochs = 20

model.fit(
    datagen.flow(x_train, y_train, batch_size=batch_size),
    steps_per_epoch=len(x_train) // batch_size,
    epochs=epochs,
    validation_data=(x_test, y_test)
)

# Evaluate the model
results = model.evaluate(x_test, y_test, batch_size=batch_size)
top_1_error = 1 - results[1]
top_5_error = 1 - results[2]

print(f"Top-1 Error Rate: {top_1_error:.4f}")
print(f"Top-5 Error Rate: {top_5_error:.4f}")
